include/seqan3/core/type_list.hpp

```diff
@@ -31,10 +35,485 @@ using type_list = meta::list<types...>;
+    else if constexpr (idx > 0)
+        return at_impl<idx - 1, tail_t...>();
+    else
+        return at_impl<sizeof...(tail_t) + 1 + idx, head_t, tail_t...>();
+}
+
+/*!\brief Implementation for seqan3::pack_traits::front.
+ * \tparam head_t   Currently viewed pack_t element.
+ * \tparam tail_t   Rest of the parameter pack.
+ * \ingroup type_traits
+ */
+template <typename head_t, typename ...tail_t>
+std::type_identity<head_t> front_impl();
+
+/*!\brief Implementation for seqan3::pack_traits::drop_front.
+ * \tparam head_t   Currently viewed pack_t element.
+ * \tparam tail_t   Rest of the parameter pack.
+ * \ingroup type_traits
+ */
+template <typename head_t, typename ...tail_t>
+type_list<tail_t...> drop_front_impl();
+
+} // namespace seqan3::pack_traits::detail
+
+// ----------------------------------------------------------------------------
+// seqan3::pack_traits
+// ----------------------------------------------------------------------------
+
+namespace seqan3::pack_traits
+{
+
+/*!\brief The size of a parameter pack.
+ * \tparam pack_t The parameter pack.
+ * \returns `sizeof...(pack_t)`
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant
+ */
+template <typename ...pack_t>
+inline constexpr size_t size = sizeof...(pack_t);
+
+/*!\brief Return the type at given index from the parameter pack.
+ * \tparam idx    The index; must be smaller than the size of the pack.
+ * \tparam pack_t The parameter pack.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * Negative indexes are supported (e.g. `at<-1, int, double, bool &>` is `bool &`).
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `idx` (linear).
+ */
+template <ptrdiff_t idx, typename ...pack_t>
+//!\cond
+    requires (idx >= 0 && idx < sizeof...(pack_t)) ||
+             (-idx <= sizeof...(pack_t))
+//!\endcond
+using at = typename decltype(detail::at_impl<idx, pack_t...>())::type;
+
+/*!\brief Return the first type from the parameter pack.
+ * \tparam pack_t The parameter pack.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <typename ...pack_t>
+//!\cond
+    requires (sizeof...(pack_t) > 0)
+//!\endcond
+using front = typename decltype(detail::front_impl<pack_t...>())::type;
+
+/*!\brief Return the last type from the parameter pack.
+ * \tparam pack_t The parameter pack.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <typename ...pack_t>
+//!\cond
+    requires (sizeof...(pack_t) > 0)
+//!\endcond
+using back = typename decltype((std::type_identity<pack_t>{}, ...))::type; // use comma operator
+
+/*!\brief Return a seqan3::type_list of the parameter pack.
+ * \tparam pack_t The parameter pack.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <typename ...pack_t>
+using concat = type_list<pack_t...>;
```

Comment:
Honestly, I don't think we need this at all. I mean that would be simply `type_list<t1, t2, t3, t4>`. Why bother to have a separate traits for this, which is basically just an alias?

Reply:
Well I wanted to keep the interfaces for list VS types very similar. 

You might have
```cpp
template <typename list1, typename list2>
```
and want to go to `concat<list1, list2>`.

Similarly:
```cpp
template <typename ...pack1, typename ...pack2>
```
and go to `concat<pack1..., pack2...>` .

Of course you could just write `type_list` for the packs, but I think there is value in consistency...

Reply:
But, there is no consistency with this, because 
```
template <typename ...pack1, typename ...pack2>
```
won't be possible if they are not already wrapped in some class. Which does not happen to often outside of the context of type_lists anyway. Also since they are in different namespaces you have to explicitly choose the traits anyway. Hence, there is no need for consistency. Either I am in the context of parameter packs or I am in the context of type_lists. Because then you could also argue that you could allow 
```
template <typename ...pack1, typename ...pack2, typename ...pack3>
...

concat<pack1..., pack2..., pack3....>
```
But this does not work with type_lists and would be more confusing then because of inconsistencies. Of course we could blow up the concat for type_lists but I think that it is rarely necessary (ok, let's see about that). So I'd like to keep these things separate and would argue that for parameter packs a concat is simply not necessary.

Reply:
Not convinced fully, but I will remove, because it's really not that important.